pennylane/ops/op_math/controlled_ops.py

```diff
@@ -139,91 +119,43 @@ def _unflatten(cls, data, metadata):
     def _primitive_bind_call(
         cls,
         base,
-        control_wires: WiresLike = "unset",
         wires: WiresLike = None,
         control_values=None,
         unitary_check=False,
         work_wires: WiresLike = (),
     ):
-        _deprecate_control_wires(control_wires)
-        work_wires = Wires(() if work_wires is None else work_wires)
-        if hasattr(base, "wires"):
-            warnings.warn(
-                "QubitUnitary input to ControlledQubitUnitary is deprecated and will be removed in v0.42. "
-                "Instead, please use a full matrix as input, or try qml.ctrl for controlled QubitUnitary.",
-                qml.PennyLaneDeprecationWarning,
-            )
-            base = base.matrix()
 
-        if control_wires == "unset":
-
-            return cls._primitive.bind(
-                base, wires=wires, control_values=control_values, work_wires=work_wires
-            )
-        # Below is the legacy interface, where control_wires provided
-        wires = Wires(() if wires is None else wires)
-
-        all_wires = control_wires + wires
+        work_wires = Wires(() if work_wires is None else work_wires)
         return cls._primitive.bind(
-            base, control_wires=all_wires, control_values=control_values, work_wires=work_wires
+            base, wires=wires, control_values=control_values, work_wires=work_wires
         )
 
     # pylint: disable=too-many-arguments,too-many-positional-arguments
     def __init__(
         self,
         base,
-        control_wires: WiresLike = "unset",
         wires: WiresLike = None,
```

Comment:
```suggestion
        wires: WiresLike,
```
Since wires are now mandatory, how about we make it mandatory? Then we can get rid of the `if not wires` check.